syntax = "proto3";

package user_service;

service UserService {
  rpc SignUp(SignUpRequest) returns (AuthReply);
  rpc Login(LoginRequest) returns (AuthReply);
  rpc LinkUser (LinkUserRequest) returns (LinkUserReply);
  rpc GetLinkedUsers (GetLinkedUsersRequest) returns (GetLinkedUsersReply);
}

message SignUpRequest {
  string email = 1;
  string password = 2;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message AuthReply {
  string uid = 1;
  string id_token = 2;
  string message = 3;
}

message LinkUserRequest {
    string patient_token = 1;
    string doctor_uid = 2;
    string relation_type = 3; // e.g. "doctor"
}

message LinkUserReply {
    string message = 1;
}

message GetLinkedUsersRequest {
    string id_token = 1;
    string relation_type = 2;
}

message RelatedUser {
    string uid = 1;
    string relation_type = 2;
}

message GetLinkedUsersReply {
    repeated RelatedUser related_users = 1;
}

